trustfall_core/src/schema/adapter/mod.rs

```diff
@@ -400,6 +408,23 @@ impl<'a> crate::interpreter::Adapter<'a> for SchemaAdapter<'a> {
                 }),
                 _ => unreachable!("unexpected edge name on type {type_name}: {edge_name}"),
             },
+            "Schema" => match edge_name.as_ref() {
+                "vertex_type" => {
+                    let candidate_destination = resolve_info.destination();
+
+                    let candidate_value_for_name =
+                        candidate_destination.statically_required_property("name");
+
+                    let mut iter = Some(self.vertex_type_iter(candidate_value_for_name));
```

Comment:
putting this into a Some() to .take it so we don't have to worry about lifetime concerns about calling into `self.` in the callback. not my best work, but atleast it works :)

Reply:
This is extremely sketchy. I register my complaint, as predicted.

I think if you make this be a free function instead of a method, that will solve the lifetime problem: the schema reference is `Copy` and lives long enough, and it won't matter whether `self` lives as long as the iterator or not.

Reply:
so this is actually right